ionic_user_interface/src/views/Gyms.vue

```diff
@@ -2,40 +2,109 @@
   <ion-page>
     <ion-content :fullscreen="true">
       <div id="container">
-        <strong>Route Maker</strong>
-        <p>Find climbing routes by gym</p>
-        <br />
-        <gym-selector v-if="canvasWidth > 0" :width="canvasWidth" />
+        <ion-row class="ion-align-items-center ion-justify-content-center">
+          <ion-col class="ion-align-self-center" size-lg="6" size-md="8" size-xs="12">
+            <div v-if="gymNameString === ''">
+              <strong>Route Maker</strong>
+              <p>Find climbing routes by gym</p>
+            </div>
+            <div v-else class="gym-name">
+              <b>-- {{ gymNameString }} --</b>
+            </div>
+          </ion-col>
+        </ion-row>
+
+        <!-- Show Gym Selector if browsing gyms, show only gym-map if individual gym -->
+        <gym-selector v-if="showGymSelector" @onGymSelect="handleOnGymSelect" />
+        <ion-row v-else class="ion-align-items-center ion-justify-content-center">
+          <ion-col class="ion-align-self-center" size-lg="6" size-md="8" size-xs="12">
+            <ion-button
+              class="ion-align-self-center ion-no-margin"
+              expand="full"
+              fill="clear"
+              color="dark"
+              @click="onClickViewMap"
+            >
+              {{ viewMap ? 'Hide Map' : 'View Map' }}
+              <ion-icon slot="end" :icon="viewMap ? map : mapOutline"></ion-icon>
+            </ion-button>
+          </ion-col>
+          <ion-col class="ion-align-self-center ion-no-padding" size-xs="12">
+            <gym-map v-if="viewMap" :gymLocation="gymLocation"></gym-map>
+          </ion-col>
+        </ion-row>
+
+        <ion-row class="ion-align-items-center ion-justify-content-center">
+          <ion-col class="ion-align-self-center" size-lg="6" size-md="8" size-xs="12">
+            <gym-route-list v-show="showGymRouteList" ref="gymRouteList" />
+          </ion-col>
+        </ion-row>
       </div>
     </ion-content>
   </ion-page>
 </template>
 
 <script lang="ts">
-import { IonContent, IonPage } from '@ionic/vue';
-import { defineComponent, onMounted, onUnmounted } from 'vue';
+import { IonContent, IonPage, IonRow, IonCol } from '@ionic/vue';
```

Comment:
![image](https://user-images.githubusercontent.com/46164638/126030863-236ab36b-b4b8-45f2-827f-faf90c15cd0e.png)

Need to import these